src/include/simeng/config/SimInfo.hh

```diff
@@ -0,0 +1,235 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+
+#include "simeng/Instruction.hh"
+#include "simeng/RegisterFileSet.hh"
+#include "simeng/arch/aarch64/ArchInfo.hh"
+#include "simeng/arch/riscv/ArchInfo.hh"
+#include "simeng/config/ModelConfig.hh"
+#include "simeng/config/yaml/ryml.hh"
+
+namespace simeng {
+namespace config {
+
+#define DEFAULT_STR "Default"
+
+/** Enum representing the possible simulation modes. */
+enum class simMode { emulation, inorder, outoforder };
```

Comment:
If this (along with the enum class for ISA type) was defined in the `simeng` namespace rather than the `simeng::config` namespace then checking against these values would be much cleaner. i.e. `... == simMode::emulation` rather than `config::simMode::emulation`

Reply:
Also, having this enum class named `SimMode` and having each enum begin with a capital letter (i.e. `Emulation`) would be more consistent with the rest of the code base

Reply:
In a similar vein as comment 1, removing SimInfo from the `config` namespace (it can still be held in the config directory) will make accessing SimInfo more readable & intuative

Reply:
The changing of simMode to SimMode and the capitalising of each entry are the only points I agree with here. 

The structure of the SimEng framework is built upon namespaces to identify and separate regions of the codebase. The folder structure is also reminiscent of this ideal. Moving SimInfo and the enum classes outside of the config namespace is not desirable as they serve the specific purpose of configuring the simulation. I don't see how the change from `config::SimMode` to `SimMode` improves cleanliness or readability and if anything, given we are using the config namespace, removing SimInfo from it would arguably make it less intuitive given the namespace usage throughout the framework.

Reply:
Thats fair enough